plugins/modules/terraform.py

```diff
@@ -473,6 +473,159 @@ def get_outputs(terraform_binary, project_path, state_file):
+    command = [
+        terraform_binary,
+        "providers",
+        "schema",
+        "-json",
+    ]  # in the command we have "providers schema", in the schema we have "provider_schemas"
+    rc, text, err = module.run_command(command, cwd=project_path)
+    if rc == 1:
+        module.warn(
+            "Could not get provider schemas. "
+            "\nstdout: {0}\nstderr: {1}".format(text, err)
+        )
+        provider_schemas = {}
+    elif rc != 0:
+        module.fail_json(
+            msg="Failure when getting provider schemas. "
+            "Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, text, err),
+            command=" ".join(command),
+        )
+    else:
+        provider_schemas = json.loads(text)
+    return provider_schemas
+
+
+def is_attribute_sensitive_in_providers_schema(provider_schemas, resource, attribute):
+    for provider_schema in provider_schemas[
+        "provider_schemas"
+    ]:
+        for resource_schema_key, resource_schema_value in (
+            provider_schemas["provider_schemas"]
+            .get(provider_schema)
+            .get("resource_schemas")
+            .items()
+        ):
+            if resource_schema_key == resource["type"]:
+                sensitive = resource_schema_value["block"]["attributes"][attribute].get(
+                    "sensitive", False
+                )  # sensitive exists only when it is True
+                return sensitive
+
+
+def is_attribute_in_sensitive_values(resource, attribute):
+    return attribute in resource["sensitive_values"]
+
+
+def filter_resource_attributes(state_file, provider_schemas):
+    # using .get() in case there is no existing .tfstate before apply
+    for resource in state_file.get("values", {}).get("root_module", {}).get("resources", {}):
+        attributes_to_remove = []
+        for attribute in resource["values"]:
+            if is_attribute_sensitive_in_providers_schema(
+                provider_schemas, resource, attribute
+            ):
+                attributes_to_remove.append(attribute)
+            elif is_attribute_in_sensitive_values(resource, attribute):
+                attributes_to_remove.append(attribute)
+        for attribute in attributes_to_remove:
+            resource["values"].pop(attribute)
+    return state_file
+
+
+def filter_outputs(state_file):
+    outputs_to_remove = []
+    # using .get() in case there is no existing .tfstate before apply
+    for output_key, output_value in state_file.get("values", {}).get("outputs", {}).items():
+        if output_value.get("sensitive", False):
+            outputs_to_remove.append(output_key)
+
+    for output in outputs_to_remove:
+        state_file["values"]["outputs"][output]["value"] = None
+
+    return state_file
+
+
+# this will not produce error if there is no init before the command
+def get_state_file_content(
+    terraform_binary, project_path, /, state_file, provider_schemas, *, sanitized
```

Comment:
```suggestion
    terraform_binary, project_path, state_file, provider_schemas, *, sanitized
```

We only want to force `sanitized` to be a keyword, but don't care whether the others are positional or keywords.